File: src/main/java/ca/weblite/jdeploy/app/services/GitHubAccountService.java
package ca.weblite.jdeploy.app.services;

import ca.weblite.jdeploy.app.exceptions.NotFoundException;
import ca.weblite.jdeploy.app.records.GitHubAccount;
import ca.weblite.jdeploy.app.repositories.GitHubAccountRepository;

import javax.inject.Inject;
import javax.inject.Singleton;
import java.util.UUID;

@Singleton
public class GitHubAccountService {
    private final GitHubAccountRepository gitHubAccountRepository;

    @Inject
    public GitHubAccountService(GitHubAccountRepository gitHubAccountRepository) {
        this.gitHubAccountRepository = gitHubAccountRepository;
    }

    public GitHubAccount findOneById(UUID uuid) throws NotFoundException {
        return gitHubAccountRepository.findOneById(uuid);
    }

    public GitHubAccount findOneByIdOrNull(UUID uuid) {
        return gitHubAccountRepository.findOneByIdOrNull(uuid);
    }
}

File: src/main/java/ca/weblite/jdeploy/app/services/GitHubAccountService.kt
package ca.weblite.jdeploy.app.services

import ca.weblite.jdeploy.app.exceptions.NotFoundException
import ca.weblite.jdeploy.app.records.GitHubAccount
import ca.weblite.jdeploy.app.repositories.GitHubAccountRepositoryInterface
import java.util.*
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class GitHubAccountService @Inject constructor(
    private val gitHubAccountRepository: GitHubAccountRepositoryInterface
) {
    @Throws(NotFoundException::class)
    fun findOneById(uuid: UUID): GitHubAccount {
        return gitHubAccountRepository.findOneById(uuid)
    }

    fun saveOne(gitHubAccount: GitHubAccount): GitHubAccount {
        return gitHubAccountRepository.saveOne(gitHubAccount)
    }
}
